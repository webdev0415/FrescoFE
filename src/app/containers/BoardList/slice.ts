import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the BoardList container
export const initialState: ContainerState = {
  loading: false,
  boardList: [],
};

const boardListSlice = createSlice({
  name: 'boardList',
  initialState,
  reducers: {
    attemptGetBoards(state, action) {
      state.loading = true;
    },
    getBoardsSuccess(state, action: PayloadAction<any>) {
      state.boardList = action.payload;
      state.loading = false;
    },
    getBoardsFail(state) {
      state.boardList = [];
      state.loading = false;
    },
    deleteBoard(state, action) {
      console.log('action', action);
      console.log('state.boardList', state.boardList);
      state.boardList = state.boardList.filter(
        board => board.id !== action.payload,
      );
    },
  },
});

export const { actions, reducer, name: sliceKey } = boardListSlice;
