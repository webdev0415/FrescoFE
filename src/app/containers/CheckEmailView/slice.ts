import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the CheckEmailView container
export const initialState: ContainerState = {};

const checkEmailViewSlice = createSlice({
  name: 'checkEmailView',
  initialState,
  reducers: {
    someAction(state, action: PayloadAction<any>) {},
  },
});

export const { actions, reducer, name: sliceKey } = checkEmailViewSlice;
