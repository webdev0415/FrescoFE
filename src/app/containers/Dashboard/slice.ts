import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the Dashboard container
export const initialState: ContainerState = {
  loading: false,
  listEmail: [],
  linkInvitation: '',
};

const dashboardSlice = createSlice({
  name: 'dashboard',
  initialState,
  reducers: {
    searchEmailRequest(state, action: PayloadAction<any>) {
      state.linkInvitation = '';
      // state.loading = true;
    },

    // TODO: define Payload type and use here
    searchEmailSuccess(state, action: PayloadAction<any>) {
      // console.log(action.payload);
      state.listEmail = action.payload;
      // state.loading = false;
    },

    searchEmailError(state, action: PayloadAction<any>) {
      // state.loading = false;
    },

    invitationRequest(state, action: PayloadAction<any>) {
      state.linkInvitation = '';
      state.loading = true;
    },

    // TODO: define Payload type and use here
    invitationSuccess(state, action: PayloadAction<any>) {
      state.linkInvitation = action.payload;
      state.loading = false;
    },

    invitationError(state, action: PayloadAction<any>) {
      state.loading = false;
    },
  },
});

export const { actions, reducer, name: sliceKey } = dashboardSlice;
