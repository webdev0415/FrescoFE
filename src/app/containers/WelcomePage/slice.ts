import { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from 'utils/@reduxjs/toolkit';
import { ContainerState } from './types';

// The initial state of the WelcomePage container
export const initialState: ContainerState = {
  loading: false,
};

const welcomePageSlice = createSlice({
  name: 'welcomePage',
  initialState,
  reducers: {
    signInRequest(state, action: PayloadAction<any>) {
      state.loading = true;
    },

    // TODO: define Payload type and use here
    signInSuccess(state, action: PayloadAction<any>) {
      state.loading = false;
    },

    signInError(state) {
      state.loading = false;
    },
  },
});

export const { actions, reducer, name: sliceKey } = welcomePageSlice;
