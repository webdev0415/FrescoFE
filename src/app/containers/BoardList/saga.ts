import { all, put, takeEvery } from 'redux-saga/effects';
import { message } from 'antd';
import { actions } from './slice';
import axios from 'axios';
import Auth from 'services/Auth';
axios.defaults.headers.common['Authorization'] = `Bearer ${Auth.getToken()}`;

// import { actions as globalActions } from '../../slice';
export function* getBoards(action) {
  const { payload } = action;

  try {
    axios.defaults.headers.common[
      'Authorization'
    ] = `Bearer ${Auth.getToken()}`;
    const response = yield axios.get(`/board/organization/${payload}`);

    yield put(actions.getBoardsSuccess(response.data));
  } catch (error) {
    yield put(actions.getBoardsFail());
    message.error(error.message);
  }
}

export function* boardListSaga() {
  yield all([takeEvery(actions.attemptGetBoards.type, getBoards)]);
}
