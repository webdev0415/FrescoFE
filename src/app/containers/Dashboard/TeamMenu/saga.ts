import { all, put, takeLatest } from 'redux-saga/effects';
import { message } from 'antd';
import axios from 'axios';
import { actions } from './slice';

export function* teamMenu(action) {
  const { payload } = action;
  const { orgId, token } = payload;
  try {
    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
    const response = yield axios.get('teams?orgId=' + orgId.orgId);
    yield put(actions.getTeamMenuSuccess(response.data));
  } catch (error) {
    yield put(
      actions.getTeamMenuError({
        message: error.message,
        status: error.response.status,
      }),
    );
    message.error(error.message);
  }
}

export function* teamMenuSaga() {
  yield all([takeLatest(actions.getTeamMenuRequest.type, teamMenu)]);
}
