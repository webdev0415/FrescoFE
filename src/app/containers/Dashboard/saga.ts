import { all, put, takeLatest } from 'redux-saga/effects';
import { message } from 'antd';
import axios from 'axios';
import { actions } from './slice';
import { v4 as uuidv4 } from 'uuid';

// import { actions as globalActions } from '../../slice';

export function* searchEmail(action) {
  const { payload } = action;
  // console.log('payload', payload);
  try {
    axios.defaults.headers.common['Authorization'] = `Bearer ${payload.token}`;
    const response = yield axios.post('/users/suggest', {
      email: payload.data.email,
      orgId: payload.data.orgId,
    });
    // console.log('response', response);
    yield put(actions.searchEmailSuccess(response.data));
  } catch (error) {
    yield put(
      actions.searchEmailError({
        message: error.message,
        status: error.response.status,
      }),
    );
    message.error(error.message);
  }
}

export function* invitation(action) {
  const { payload } = action;
  const { history, permission, token, orgId, data } = payload;
  const feToken = uuidv4();
  // console.log('payload', data, payload?.data[0], feToken);
  try {
    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
    const response = yield axios.post('invitation', {
      toUserId: data[0]?.id || '',
      toEmail: data[0]?.email || data.email,
      token: feToken,
      permission,
      orgId,
    });
    // console.log('response', response);
    yield put(actions.invitationSuccess(response.data));
    message.success('Invitation successfully.');
  } catch (error) {
    history.push('/err');
    yield put(
      actions.invitationError({
        message: error.message,
        status: error.response.status,
      }),
    );
    message.error(error.message);
  }
}

export function* dashboardSaga() {
  yield all([
    takeLatest(actions.searchEmailRequest.type, searchEmail),
    takeLatest(actions.invitationRequest.type, invitation),
  ]);
}
