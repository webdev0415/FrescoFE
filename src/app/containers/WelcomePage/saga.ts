import { all, put, takeLatest } from 'redux-saga/effects';
import { message } from 'antd';
import axios from 'axios';

import { actions } from './slice';
import { actions as globalActions } from '../../slice';

export function* signIn(action) {
  try {
    const { payload } = action;
    const { token, history } = payload;
    axios.defaults.headers.common['Authorization'] = `Bearer ${token}`;
    const response = yield axios.get('auth/me');
    yield put(actions.signInSuccess(response.data));
    const authInfo = {
      user: response.data,
      token: { accessToken: token },
    };
    yield put(globalActions.setAuthInformation(authInfo));
    localStorage.setItem('authInformation', JSON.stringify(authInfo));
    message.success('Logged in successfully.');
    history.push('/auth/welcome-page');
  } catch (error) {
    message.error(error.message);
    yield put(actions.signInError());
  }
}

export function* welcomePageSaga() {
  yield all([takeLatest(actions.signInRequest.type, signIn)]);
}
