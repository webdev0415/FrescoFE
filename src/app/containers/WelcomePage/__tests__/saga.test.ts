import { all, put, takeLatest } from 'redux-saga/effects';
import { actions } from '../slice';
import { actions as globalActions } from '../../../slice';

import { signIn, welcomePageSaga } from '../saga';

describe('SignIn-> Saga-> signIn', () => {
  let signInSagaIterator: ReturnType<typeof signIn>;
  beforeEach(() => {
    const token = '123';
    const action = actions.signInRequest({
      history: { push: () => {} },
      token,
    });
    signInSagaIterator = signIn(action);
    const { payload } = action;

    const requestDescriptor = signInSagaIterator.next(payload).value;

    expect(requestDescriptor).toMatchSnapshot();
  });

  it('signIn success', () => {
    const successData = {
      user: {
        username: 'tariq',
        role: 'USER',
        email: 'example@fresco.com',
      },
      token: {
        accessToken: '123',
      },
    };

    const putDescriptorSignInSuccess = signInSagaIterator.next({
      data: successData.user,
    }).value;
    expect(putDescriptorSignInSuccess).toEqual(
      put(actions.signInSuccess(successData.user)),
    );

    const putDescriptorSetAuthInformation = signInSagaIterator.next({
      data: successData.user,
    }).value;
    expect(putDescriptorSetAuthInformation).toEqual(
      put(globalActions.setAuthInformation(successData)),
    );
    const iteration = signInSagaIterator.next();
    expect(iteration.done).toBe(true);
    expect(localStorage.getItem('authInformation')).toBe(
      JSON.stringify(successData),
    );
  });

  it('signIn error', () => {
    const error = {
      message: 'Not found',
    };
    const putDescriptorSignInError = signInSagaIterator.throw(error).value;

    expect(putDescriptorSignInError).toEqual(put(actions.signInError()));
    const iteration = signInSagaIterator.next();
    expect(iteration.done).toBe(true);
  });
});

describe('SignIn-> Saga-> signInSaga', () => {
  it('signInSaga watch all', () => {
    let signInSagaWatchAllIterator: ReturnType<typeof welcomePageSaga>;
    signInSagaWatchAllIterator = welcomePageSaga();
    const watchDescriptor = signInSagaWatchAllIterator.next().value;
    expect(watchDescriptor).toMatchSnapshot();
  });
});
