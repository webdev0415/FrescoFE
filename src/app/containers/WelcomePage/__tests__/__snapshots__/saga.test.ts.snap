// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn-> Saga-> signIn signIn error 1`] = `Promise {}`;

exports[`SignIn-> Saga-> signIn signIn success 1`] = `Promise {}`;

exports[`SignIn-> Saga-> signInSaga signInSaga watch all 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": true,
  "payload": Array [
    Object {
      "@@redux-saga/IO": true,
      "combinator": false,
      "payload": Object {
        "args": Array [
          "welcomePage/signInRequest",
          [Function],
        ],
        "context": null,
        "fn": [Function],
      },
      "type": "FORK",
    },
  ],
  "type": "ALL",
}
`;
